src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -154,28 +154,50 @@ void VehicleInfoPlugin::UnsubscribeFromRemovedVDItems() {
   application_manager_->GetRPCService().ManageHMICommand(message);
 }
 
+bool IsOtherAppAlreadySubscribedFor(
+    const std::string& ivi_name,
+    application_manager::ApplicationManager& app_mngr,
+    application_manager::Application& current_app) {
+  auto applications = app_mngr.applications();
+
+  for (auto& app : applications.GetData()) {
+    auto& ext = VehicleInfoAppExtension::ExtractVIExtension(*app);
+    if (ext.isSubscribedToVehicleInfo(ivi_name) &&
+        (app->app_id() != current_app.app_id())) {
+      return true;
+    }
+  }
+  return false;
+}
+
 void VehicleInfoPlugin::ProcessResumptionSubscription(
     application_manager::Application& app, VehicleInfoAppExtension& ext) {
   LOG4CXX_AUTO_TRACE(logger_);
   smart_objects::SmartObject msg_params =
       smart_objects::SmartObject(smart_objects::SmartType_Map);
   msg_params[strings::app_id] = app.app_id();
+  size_t initial_length = msg_params.length();
   const auto& subscriptions = ext.Subscriptions();
+
   if (subscriptions.empty()) {
     LOG4CXX_DEBUG(logger_, "No vehicle data to subscribe. Exiting");
     return;
   }
 
-  for (const auto& item : subscriptions) {
-    msg_params[item] = true;
+  for (auto& ivi : subscriptions) {
```

Comment:
move initial_length closer to this loop.

Reply:
Done in commit 7571f74